### ISSUE TITLE
Allow shared clipboard in GUI Linux Apple VMs on macOS 13

### ISSUE THREAD

**Issue by conath** [utmapp/UTM#4121]
macOS 13 brings shared clipboard support to [Apple-backend GUI Linux VMs](https://developer.apple.com/documentation/virtualization/running_gui_linux_in_a_virtual_machine_on_a_mac). It looks like a small change to UTM is necessary to enable this feature.

It would be great if the shared clipboard could be toggled from the VM settings.

It’s worth noting the updated Apple Virtualization backend uses SPICE for this feature, just like UTM/QEMU does. :)

**Comment by vincent-olivert-riera** [comment]
> It’s worth noting the updated Apple Virtualization backend uses **SPICE** for this feature, just like UTM/QEMU does. :)

Bad news for Wayland then :-(

- https://gitlab.freedesktop.org/spice/linux/vd_agent/-/issues/26
- https://unix.stackexchange.com/questions/698083/kvm-clipboard-sharing-with-client-running-wayland
- https://www.reddit.com/r/swaywm/comments/pg0rqi/clipboard_sharing_using_spicevdagent_not_working/

**Comment by git0uttaHereIci** [comment]
> macOS 13 brings shared clipboard support to [Apple-backend GUI Linux VMs](https://developer.apple.com/documentation/virtualization/running_gui_linux_in_a_virtual_machine_on_a_mac). It looks like a small change to UTM is necessary to enable this feature.
> 
> It would be great if the shared clipboard could be toggled from the VM settings.
> 
> It’s worth noting the updated Apple Virtualization backend uses SPICE for this feature, just like UTM/QEMU does. :)

can you rephrase this for a complete baphoon?

**Comment by git0uttaHereIci** [comment]
hey Boss, So how can I get shared clipboard and file sharing to work..?

step 1
step B
step 3

etc.../ is failing me

**Comment by conath** [comment]
@git0uttaHereIci This issue is about a feature in macOS 13 that *will* allow shared clipboard to work for more VMs, once UTM is updated to use it.

**Comment by git0uttaHereIci** [comment]
> hey Boss, So how can I get shared clipboard and file sharing to work..?
> 
> step 1 step B step 3
> 
> etc.../ is failing me

https://mac.getutm.app/support/

**Comment by git0uttaHereIci** [comment]
> > It’s worth noting the updated Apple Virtualization backend uses **SPICE** for this feature, just like UTM/QEMU does. :)
> 
> Bad news for Wayland then :-(
> 
> * https://gitlab.freedesktop.org/spice/linux/vd_agent/-/issues/26
> * https://unix.stackexchange.com/questions/698083/kvm-clipboard-sharing-with-client-running-wayland
> * https://www.reddit.com/r/swaywm/comments/pg0rqi/clipboard_sharing_using_spicevdagent_not_working/

https://mac.getutm.app/support/

**Comment by peterneutron** [comment]
@conath Sorry to use this closed issue but I wanted to be sure before opening a new one.

I skimmed trough commit 61d05165ef49cfdd87ce9f6dee6d10f5ae9df82e, as I understand it there should be a checkbox to enable clipboard sharing for Apple-backend GUI VMs on supported configurations. Does this not apply for `macOS 13.x` guests using `UTM 4.0.7 RC`? Or am I just missing something.

**Comment by conath** [comment]
@osy ?

**Comment by osy** [comment]
It’s only supported on Linux guests as macos does not have the drivers yet

**Comment by peterneutron** [comment]
@osy Thanks for the clarification, I thought `macOS 13.x` would expose those APIs/interfaces by default because they support it on Linux guests. Is there any Apple Developer announcements that they're going to support this natively or are we depending on 3rd parties e.g. Spice for driver support?

**Comment by osy** [comment]
Not that I’m aware of
